v8go.cc

```diff
@@ -26,12 +27,14 @@ const int ScriptCompilerEagerCompile = ScriptCompiler::kEagerCompile;
 
 struct m_ctx {
   Isolate* iso;
-  std::vector<m_value*> vals;
+  std::unordered_map<long, m_value*> vals;
   std::vector<m_unboundScript*> unboundScripts;
   Persistent<Context> ptr;
+  long nextValId;
 };
 
 struct m_value {
+  long id;
```

Comment:
I think we need a constructor here to enforce that this member is initialized to zero (eg, `m_value() : id(0) { }`). Otherwise, we have to depend on the caller to invoke like `new m_value()` instead of `new m_value` and everybody here is doing the latter. The reason I think this is that we explicitly check whether id is zero further down.